src/bp-members/classes/class-bp-members-signups-feature.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * BuddyPress Member's notice feature Class.
+ *
+ * @package BuddyPress
+ * @subpackage Members
+ * @since 15.0.0
+ */
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * The BP Signups feature is now optional.
+ *
+ * If you want to disable it, you can use:
+ * `add_filter( 'bp_is_members_signups_active', '__return_false' );`
+ *
+ * @since 15.0.0
+ */
+class BP_Members_Signups_Feature extends BP_Component_Feature {
+
+	/**
+	 * Signups Feature initialization.
+	 *
+	 * @since 15.0.0
+	 */
+	public function __construct() {
+		parent::init( 'signups', 'members' );
+	}
+
+	/**
+	 * Set up Signups feature global variables.
+	 *
+	 * @since 15.0.0
+	 */
+	public function globals() {
+		buddypress()->signup = new stdClass();
+
+		parent::globals();
+	}
+
+	/**
+	 * Include Signups feature files.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @see `BP_Component_Feature::includes()` for description of parameters.
+	 *
+	 * @param array $includes See {@link BP_Component_Feature::includes()}.
+	 */
+	public function includes( $includes = array() ) {
+		parent::includes( array( 'bp-members-signups' ) );
+	}
+
+	/**
+	 * Include screen/action files later & when on specific pages.
+	 *
+	 * @since 15.0.0
+	 */
+	public function late_includes() {
+		// Bail if PHPUnit is running.
+		if ( defined( 'BP_TESTS_DIR' ) ) {
+			return;
+		}
```

Comment:
curious as to why?

Reply:
I believe it's simply to avoid loading these files when running Unit tests. It's in place since: https://buddypress.trac.wordpress.org/ticket/7218

Reply:
I just removed this check, I agree it's not needed!